art_internal_impl.hpp

```diff
@@ -550,18 +604,50 @@ class basic_inode_impl : public ArtPolicy::header_type {
   using critical_section_policy =
       typename ArtPolicy::template critical_section_policy<T>;
 
+  using lock_policy = typename ArtPolicy::lock_policy;
+  using read_critical_section = typename ArtPolicy::read_critical_section;
+  
   using db_leaf_unique_ptr = typename ArtPolicy::db_leaf_unique_ptr;
 
   using db = typename ArtPolicy::db;
 
-  // The first element is the child index in the node, the 2nd one is pointer
-  // to the child. If not present, the pointer is nullptr, and the index
-  // is undefined. The pointer element is indirect to support inserts needing to
-  // know the location of the child node in case it needs to be replaced in the
-  // parent.
+  // The first element is the child index in the node, the second
+  // element is a pointer to the child.  If there is no such child,
+  // the pointer is nullptr, and the child_index is undefined.
   using find_result =
-      std::pair<std::uint8_t, critical_section_policy<node_ptr> *>;
-
+      std::pair< std::uint8_t // child_index
+               , critical_section_policy<node_ptr> *  // child
+               >;
+
+  // A tuple that is returned by the iterator visitation pattern which
+  // represents the path in the tree for an internal node.
+  //
+  // Note: The node pointer is either an internal node or a leaf.
+  //
+  // Note: The key byte is the byte from the key that was consumed as
+  // you step down to the child node. This is the same as the child
+  // index (if you convert std::uint8_t to std::byte) for N48 and
+  // N256, but it is different for N4 and N16 since they use a sparse
+  // encoding of the keys.  It is represented explicitly to avoid
+  // searching for the key byte in the N48 and N256 cases.
+  //
+  // Note: The child_index is the index of the child node within that
+  // internal node (except for N48, where it is the index into the
+  // child_indexes[] and is in fact the same data as the key byte).
+  // Overflow for the child_index can only occur for N48 and N256.
```

Comment:
TODO why the difference between N48-N256 and N4-N16 here

Reply:
This is just how the data structure is.  The N4/N16 is a list of only the used key bytes. The N48/N256 is an array of all possible key bytes